File: app/src/main/java/xyz/iridiumion/enlightened/highlightingdefinitions/HighlightingDefinitionLoader.java
package xyz.iridiumion.enlightened.highlightingdefinitions;

import xyz.iridiumion.enlightened.editor.HighlightingDefinition;
import xyz.iridiumion.enlightened.highlightingdefinitions.definitions.GenericHighlightingDefinition;
import xyz.iridiumion.enlightened.highlightingdefinitions.definitions.JavaScriptHighlightingDefinition;
import xyz.iridiumion.enlightened.highlightingdefinitions.definitions.NoHighlightingDefinition;

/**
 * Author: 0xFireball
 */
public class HighlightingDefinitionLoader {

    public HighlightingDefinition selectDefinitionFromFileExtension(String selectedFileExt) {
        switch (selectedFileExt) {
            case "js":
                return new JavaScriptHighlightingDefinition();
            case "txt":
                return new NoHighlightingDefinition();
            default:
                return new GenericHighlightingDefinition();
        }
    }
}

File: app/src/main/java/xyz/iridiumion/enlightened/highlightingdefinitions/HighlightingDefinitionLoader.kt
package xyz.iridiumion.enlightened.highlightingdefinitions

import xyz.iridiumion.enlightened.editor.HighlightingDefinition
import xyz.iridiumion.enlightened.highlightingdefinitions.definitions.*

/**
 * Author: 0xFireball
 */
class HighlightingDefinitionLoader {

    fun selectDefinitionFromFileExtension(selectedFileExt: String): HighlightingDefinition {
        when (selectedFileExt) {
            "js" -> return JavaScriptHighlightingDefinition()
            "java" -> return JavaHighlightingDefinition()
            "cs" -> return CSharpHighlightingDefinition()
            "cpp", "cxx" -> return CPlusPlusHighlightingDefinition()
            "txt" -> return NoHighlightingDefinition()
            else -> return GenericHighlightingDefinition()
        }
    }
}
